x/evmutil/keeper/conversion_evm_native.go

```diff
@@ -102,13 +110,32 @@ func (k Keeper) ConvertERC20ToCoin(
 		return err
 	}
 
+	amountToLock := amount.BigInt()
+	amountToMint := amount.BigInt()
+
+	// handle bep3 conversion pair (18 decimals erc20 to 8 decimals sdk.Coin)
+	// When converting bep3 erc20 assets to sdk.Coin, the following will happen
+	// 1. Convert provided 18 decimals erc20 amount to 8 decimals sdk amount, then mint those.
+	//    Dust from converting 18 decimals to 8 decimals are ignored and will
+	//    remain in the original wallet.
+	// 2. Convert to-be-minted 8 decimals sdk.Coin back to 18 decimals, then lock those.
+	if IsEvmNativeBep3Conversion(pair) {
+		amountToMint, _ = ConvertBep3ERC20AmountToCoinAmount(amount.BigInt())
+
+		// make sure we have at least 1 sdk.Coin to mint
+		if amountToMint.Cmp(big.NewInt(0)) == 0 {
```

Comment:
good catch on this edge case, would be good to cover in a test (in this or levi's branch)